packages/core/src/actions.ts

```diff
@@ -330,22 +334,23 @@ export function log<TContext, TEvent extends EventObject>(
     type: actionTypes.log,
     label,
     expr
-  };
+  } as LogAction<TContext, TEvent>;
 }
 
-export const resolveLog = <TContext, TEvent extends EventObject>(
+export function resolveLog<TContext, TEvent extends EventObject>(
   action: LogAction<TContext, TEvent>,
   ctx: TContext,
   _event: SCXML.Event<TEvent>
-): LogActionObject<TContext, TEvent> => ({
-  // TODO: remove .expr from resulting object
```

Comment:
should the `expr` property stay on the resolved action?

Reply:
I don't think so; the resulting action object is the result of the dynamic action; `.expr` is an implementation detail.

Reply:
Since we are spreading the `action` here - the `expr` becomes part of it. So if this doesn't have to stay on the resolved action it might be worth removing it.